src/Utilities/VectorInt.f90

```diff
@@ -0,0 +1,143 @@
+module VectorIntModule
+  use KindModule, only: I4B
+  use SimModule, only: ustop
+  use ArrayHandlersModule, only: ExpandArray
+  implicit none
+  private
+  public :: VectorInt  
+  
+  integer(I4B), parameter :: defaultInitialCapacity = 4
+  
+  ! This is a dynamic vector type for integers
+  type :: VectorInt
+    integer(I4B), private, allocatable :: values(:) ! the internal array for storage
+    integer(I4B) :: size                            ! the number of elements (technically this stuff should be unsigned)
```

Comment:
The intention for this type is to use it to introduce dynamic memory when building up the interface model. Once built, it can be deleted again and it need not be in the memory manager. I figured that 'allocatable' would be a bit more efficient in such a case.